### DISCUSSION TITLE
Installing signal-cli and scli within QubesOs

### DISCUSSION THREAD

**Discussion by grif0389** [AsamK/signal-cli#1134] [General]
Hello can I start by thanking those that take the time to read this message.

Since becoming more security conscience I'm trying to use Qubes Os for all my communications.  I wish to use signal on my computer without a phone, so that I can still interact with normal people.

Ive spent a good few days now working on installing signal-cli and cli in Qubes OS and researching on the web and I'm running into a problem.

Here's what I'm doing;
#Installed dependencies  for signal-cli and cli in a whonix ws template vm.
#Created AppVM from the whonix ws template.
#Installed signal-cli and cli in the /home/user directory of the AppVM.
#Registered an account with signal using a sim in the cloud service.

I can both send and receive messages using the cli terminal user interface,  I can also receive and send messages directly using signal-cli -a ACCOUNT receive command.  

The problem is as soon as I close the terminal window and re-open, both the sent and the received messages are gone.  Its seems that signal-cli is sending and receiving all messages, however it isn't saving the conversations.  

I suspect that this is a case of user error, the solution is probably very simple, that being said I'm stuck!

Any help would be greatly appreciated.

Many thanks.

:)

**Comment by exquo** [comment]
> signal-cli isn't saving the conversations

Yes, singal-cli does not save the messages. They need to be written to the disk manually, e.g. by redirecting the signal-cli's output to a file.

`scli` has the option to save the messages history. It can be turned on by supplying a `--save-history` (or `-s`) flag to `scli`.
